### PR TITLE
improve the current pod in accurary

### PR DESCRIPTION
**Pull request by DeliangFan**
Current Pod = Waited Pod + Binded Pod

### CODE REVIEW

File: pkg/coscheduling/coscheduling.go [hunk 1]
```diff
@@ -164,20 +164,19 @@ func (cs *Coscheduling) Permit(ctx context.Context, state *framework.CycleState,
 	}
 
 	namespace := p.Namespace
-	// TODO get actually scheduled(bind successfully) account from the SharedLister
-	running := cs.calculateRunningPods(podGroupName, namespace)
+	binded := cs.calculateBindedPods(podGroupName, namespace)
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling.go:167]: Hmm... I don't quite think `bound` is better than `running`. Usually `running` means a Pod admitted by kubelet already, while `bound` just means the Pod's `.spec.nodeName` is already set.

File: pkg/coscheduling/coscheduling.go [hunk 2]
```diff
@@ -233,10 +232,17 @@ func (cs *Coscheduling) calculateTotalPods(podGroupName, namespace string) int {
...
-			return true
+		if pod.Labels[PodGroupName] == podGroupName && pod.Namespace == namespace {
+			if pod.Status.Phase == v1.PodRunning {
+				return true
+			}
+			for _, c := range pod.Status.Conditions {
+				if c.Type == v1.PodScheduled || c.Type == v1.PodInitialized || c.Type == v1.ContainersReady || c.Type == v1.PodReady {
+					return true
+				}
+			}
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling.go:245]: Why this is needed?

Comment (DeliangFan) [reply pkg/coscheduling/coscheduling.go:245]: - running := cs.calculateRunningPods(podGroupName, namespace)
It only calculate the pods has bound to a node and all of the containers have been started, which may miss these bound pod whose PodPhase is still in Pending before containers start during pull images.

Is my opinion correct？

Comment (Huang-Wei) [reply pkg/coscheduling/coscheduling.go:245]: That's correct - running means at least one of the containers is already running.

Given that the core scheduler doesn't really look at the phase of a Pod, @denkensk do you think we should just check a Pod's spec.nodeName here?

Comment (DeliangFan) [reply pkg/coscheduling/coscheduling.go:245]: After a deep dive into the scheduler framework, I also believe that check spec.nodeName is much better than the Status。Conditions

Comment (denkensk) [reply pkg/coscheduling/coscheduling.go]: Thanks @Huang-Wei  I agree to just check a Pod's nodeName.

Comment (denkensk) [reply pkg/coscheduling/coscheduling.go]: Thanks @DeliangFan
If we just check the Pod's `spec.nodeName`,  we need to rename `calculateRunningPods `  to `calculateBindedPods`.

Comment (DeliangFan) [reply pkg/coscheduling/coscheduling.go]: Done, please review again

**Pull request merged**